lib/hammer_cli_csv/host_groups.rb

```diff
@@ -45,66 +47,76 @@ def export
+              foreman_organization(:name => organization)
+            end,
+            'location_ids' => collect_column(line[LOCATIONS]) do |location|
+              foreman_location(:name => location)
+            end
+          }
+        }
+
         count(line[COUNT]).times do |number|
           name = namify(line[NAME], number)
+          params['hostgroup']['name'] = name
+
           if !@existing.include? name
             print "Creating host group '#{name}'..." if option_verbose?
-            # @api.resource(:hosts).call(:create, {
-            #     'host' => {
-            #       'name' => name,
-            #       'root_pass' => 'changeme',
-            #       'mac' => namify(line[MACADDRESS], number),
-            #       'organization_id' => foreman_organization(:name => line[ORGANIZATION]),
-            #       'location_id' => foreman_location(:name => line[LOCATION]),
-            #       'environment_id' => foreman_environment(:name => line[ENVIRONMENT]),
-            #       'operatingsystem_id' => foreman_operatingsystem(:name => line[OPERATINGSYSTEM]),
-            #       'architecture_id' => foreman_architecture(:name => line[ARCHITECTURE]),
-            #       'domain_id' => foreman_domain(:name => line[DOMAIN]),
-            #       'ptable_id' => foreman_partitiontable(:name => line[PARTITIONTABLE])
-            #     }
-            # })
+            hostgroup = @api.resource(:hostgroups).call(:create, params)
+            @existing[name] = hostgroup['id']
           else
             print "Updating host '#{name}'..." if option_verbose?
-            # @api.resource(:hosts).call(:update, {
-            #     'id' => @existing[name],
-            #     'host' => {
-            #       'name' => name,
-            #       'mac' => namify(line[MACADDRESS], number),
-            #       'organization_id' => foreman_organization(:name => line[ORGANIZATION]),
-            #       'environment_id' => foreman_environment(:name => line[ENVIRONMENT]),
-            #       'operatingsystem_id' => foreman_operatingsystem(:name => line[OPERATINGSYSTEM]),
-            #       'architecture_id' => foreman_architecture(:name => line[ARCHITECTURE]),
-            #       'domain_id' => foreman_domain(:name => line[DOMAIN]),
-            #       'ptable_id' => foreman_partitiontable(:name => line[PARTITIONTABLE])
-            #     }
-            # })
+            params['id'] = @existing[name]
+            hostgroup = @api.resource(:hostgroups).call(:update, params)
           end
+
+          # TODO: puppet classes
+          puppetclass_ids = collect_column(line[PUPPET_CLASSES]) do |puppet_class|
+            module_name, name = puppet_class.split('/')
+            foreman_puppet_class(:name => name)
+          end
+          existing_ids = hostgroup['puppet_classes'].collect { |puppet_class| puppet_class['id'] }
+          # DELETE existing_ids - puppetclass_ids
+          # POST puppetclass_ids - existing_ids
```

Comment:
Export/import host groups is pretty complicated and will remain in the "unsupported" category for a while. The TODO comment above will remain unfixed in this PR and the two comments are what needs to happen. I think. Not all the foreman APIs accept adding relationships versus fully resetting them.

tl;dr - more work to do in future